ports/qemu-riscv/entry.s

```diff
@@ -0,0 +1,51 @@
+/*
+
+Adapted from https://wiki.osdev.org/RISC-V_Bare_Bones#entry.S
+
+According to the page history, this content was added after 20110703, and
+following the text found at https://wiki.osdev.org/OSDev_Wiki:License the
+overall licence for this code is as follows:
```

Comment:
We have to be very careful with licensing, this kind of file can easily find itself copied to other parts of the code base, and the license will follow the code.

It's a pretty simple file.  Could it be written from scratch so that the licensing can be 100% clear?

Reply:
No problem, I already have another version of this that adds stack canaries and other bits and that's written from scratch (as the linkerscript changed).